custom_components/birthdays/__init__.py

```diff
@@ -45,28 +51,48 @@ async def async_setup(hass, config):
     tasks = [asyncio.create_task(device.update_data()) for device in devices]
     await asyncio.wait(tasks)
 
+    _LOGGER.debug(devices)
+
     return True
 
 
 class BirthdayEntity(Entity):
 
-    def __init__(self, name, date_of_birth, icon, hass):
+    def __init__(self, name, date_of_birth, unique_id, icon, attributes, hass):
         self._name = name
-        self._date_of_birth = date_of_birth
+
+        if unique_id is not None:
+            self._unique_id = slugify(unique_id)
+        else: 
+            self._unique_id = slugify(name)
+
         self._icon = icon
+        self._date_of_birth = date_of_birth
         self._age_at_next_birthday = 0
```

Comment:
Since this is moved into the _extra_state_attributes, this member can be removed

Reply:
Removed